extensions/sql-database-projects/src/sqldbproj.d.ts

```diff
@@ -0,0 +1,32 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+declare module 'sqldbproj' {
+	import * as vscode from 'vscode';
+	export const enum extension {
+		name = 'Microsoft.sql-database-projects'
+	}
+
+	/**
+	 * sql database projects extension
+	 */
+	export interface IExtension {
+		/**
+		 * Create a project
+		 * @param name name of the project
+		 * @param location the parent directory
+		 * @param projectTypeId the ID of the project/template
+		 * @returns Uri of the newly created project file
+		 */
+		createProject(name: string, location: vscode.Uri, projectTypeId: string): Promise<vscode.Uri>;
+
+		/**
+		* Adds the list of files and directories to the project, and saves the project file
+		* @param projectFile The Uri of the project file
+		* @param list list of file and folder paths to add. Files and folders must already exist
```

Comment:
I wasn't sure what the behavior should be if a file/folder passed in doesn't exist. Should other files/folders not be added if any in the list doesn't exist? Or should the other files/folders in the list still be added even if another one doesn't exist?

Right now, the error only shows in the console and if a file/folder doesn't exist, then the other files/folders after that one in the list do not get added.

Reply:
I think it is fair to require all the files exists, if not, none will be added.